app/models/roster/creator.rb

```diff
@@ -74,18 +74,33 @@ def perform
     rescue Result::Error, ActiveRecord::ActiveRecordError => err
       Result.failed(@roster, err.message)
     end
-    # rubocop:enable Metrics/MethodLength
 
     private
 
     def add_identifiers_to_roster(raw_identifiers_string, google_ids: [])
-      identifiers = raw_identifiers_string.split("\r\n").reject(&:blank?).uniq
+      identifiers = raw_identifiers_string.split("\r\n").reject(&:blank?)
 
       identifiers.zip(google_ids).each do |identifier, google_user_id|
-        @roster.roster_entries << RosterEntry.new(identifier: identifier, google_user_id: google_user_id)
+        identifier = check_for_duplicates(@roster.roster_entries, identifier)
+
+        @roster.roster_entries << RosterEntry.new(
+          roster: @roster, identifier: identifier, google_user_id: google_user_id
+        )
+        @roster.save!
       end
     end
 
+    def check_for_duplicates(roster_entries, identifier)
```

Comment:
Are you able to bring this logic together in one place instead of duplicating in two places?

I think a good interface would be a method that takes a list of identifiers, and outputs a list of identifiers with duplicates given a suffix. Then we could call this on line 82 like

```ruby
identifiers = add_suffix_to_duplicates(identifiers)
```

Reply:
I'll look into this 👍